lib/dpl/provider/pypi.rb

```diff
@@ -32,23 +32,13 @@ def skip_upload_docs?
           (options.has_key?(:skip_upload_docs) && options[:skip_upload_docs])
       end
 
-      def self.install_setuptools
-        shell 'wget https://bootstrap.pypa.io/ez_setup.py -O - | sudo python'
-        shell 'rm -f setuptools-*.zip'
-      end
-
-      def self.install_twine
-        shell("pip install twine", retry: true) if `which twine`.chop.empty?
-      end
-
-      def initialize(*args)
-        super(*args)
-        self.class.pip 'wheel' if pypi_distributions.to_s.include? 'bdist_wheel'
+      def install_deploy_dependencies
+        unless context.shell "wget -O - https://bootstrap.pypa.io/get-pip.py | python - --no-setuptools --no-wheel && " \
+                             "pip install --upgrade setuptools twine wheel"
```

Comment:
Previously, we were using `sudo` to run `ez_setup.py`. https://github.com/travis-ci/travis-ci/issues/9139 reports that this perhaps leads to "permission denied" error. Is it feasible to do the same here?

Reply:
Ideally, this should run in the same virtualenv where the tests were running. I believe, this happens to `language: python`. Is it possible to have this in other envs? It is a bad practice to influence system interpreter installation. `dpl` is being run in envs other than Travis. It would make me feel terribly uncomfortable, if such deployment tool would ask me to execute smth as root. I think, this should be at least limited to user dir.

Reply:
If it's really required to be run as root, why not run the whole thing with sudo? This would be explicit and will be run by the user mindfully.